frontend/src/tests/application-tests.js

```diff
@@ -1,24 +1,391 @@
+
+        // Get the current active user
+        let resp = await apiGET("/debug/active_user");
+        expect(resp).toHaveStatus("success");
+        defaultUser = resp.payload;
+
+        // Making sure the user is the seeded applicant before this get request for the purposes of the next test
+        // Created a user for the applicant to make sure they exist before changing the default user to applicant
+        Object.assign(userCreatedFromApplicant, {
+            utorid: applicant.utorid,
+            roles: ["ta"],
+        });
+
+        resp = await apiPOST("/debug/users", userCreatedFromApplicant);
+        expect(resp).toHaveStatus("success");
+        Object.assign(userCreatedFromApplicant, resp.payload);
+
+        // Create a new user with TA permissions
+        resp = await apiPOST("/debug/users", userWithTaPermissions);
+        expect(resp).toHaveStatus("success");
+        Object.assign(userWithTaPermissions, resp.payload);
     }, 30000);
 
     // These tests set data through the `/public/postings` route,
     // but read data through the `/api/v1/admin` route.
     describe("Public route tests", () => {
-        it.todo("Get survey.js posting data through public route");
+        it("Get survey.js posting data through public route", async () => {
+            // Make sure the user has admin permissions before this post request
+            // Make a new posting and update <posting> to include the id of the posting
+            let resp = await apiPOST("/admin/postings", {
+                ...posting,
+                session_id: session.id,
+            });
+            expect(resp).toHaveStatus("success");
+            checkPropTypes(postingPropTypes, resp.payload);
+            expect(resp.payload.id).not.toBeNull();
+            Object.assign(posting, resp.payload);
+
+            await switchToApplicantUser();
+
+            // read survey.js posting data
+            resp = await apiGET(`/public/postings/${posting.url_token}`, true);
+            expect(resp).toHaveStatus("success");
+            checkPropTypes(surveyPropTypes, resp.payload);
+            surveyPrefill = resp.payload["prefilled_data"];
+
+            await restoreDefaultUser();
+        });
+
+        // broken up into 2 tests: applicant & application(not yet implemented in the backend)
+        it("Survey.js posting data is pre-filled based on prior applicant", async () => {
+            // this test depends on the previous test on the terms of surveyPrefill variable
+            for (const prefilledKey of [
+                "utorid",
+                "student_number",
+                "first_name",
+                "last_name",
+                "email",
+                "phone",
+            ]) {
+                // check if the applicant prefill information is correct
+                expect(surveyPrefill[prefilledKey]).toEqual(
+                    applicant[prefilledKey]
+                );
```

Comment:
What do you think of 

```
expect(surveyPrefill[prefilledKey]).toEqual(
    expect.objectContaining(applicantInfo))
```
or

```
expect(surveyPrefill[prefilledKey]).toEqual(
    expect.objectContaining(applicant))
```
This would give you a diff instead of just the first property it failed at! Alternatively, you could write a

`it.each` for each property, to at least have a list of all properties that were not prefilled, but personally, I like the `objectContaining` approach better.

Reply:
I feel like this is one of the right ways to do implement the test since we are testing if such specific fields are prefilled.

Reply:
It's just a style/usability suggestion, if you don't like it - no worries :)